src/module/pageSettings.ts

```diff
@@ -35,23 +36,40 @@ export const pageSettingsSlice = createSlice({
       state.searchParams = action.payload;
       return state;
     },
+    setLoaded(state, action: PayloadAction<boolean>) {
+      state.loaded = action.payload;
+      return state;
+    },
   },
   initialState: { ...defaultPageSettings, page: '', searchParams: {} } as PageSettingsSlice,
 });
 
-const { setPageSettings, setPageParams } = pageSettingsSlice.actions;
+const { setPageSettings, setPageParams, setLoaded } = pageSettingsSlice.actions;
 export { setPageParams };
 
 export function usePageSettings(): PageSettingsSlice;
-export function usePageSettings(settings: Partial<PageSettings>, deps?: DependencyList): void;
-export function usePageSettings(settings?: Partial<PageSettings>, deps: DependencyList = []): PageSettingsSlice | void {
+export function usePageSettings(
+  settings: Partial<PageSettings & { instantLoading: boolean }>,
+  deps?: DependencyList,
+): void;
+export function usePageSettings(
+  settings?: Partial<PageSettings & { instantLoading: boolean }>,
+  deps: DependencyList = [],
+): PageSettingsSlice | void {
   /* eslint-disable react-hooks/rules-of-hooks */
   const pathname = usePathname();
   if (settings) {
     const dispatch = useAppDispatch();
     useEffect(() => {
       dispatch(((dispatch: AppDispatch) =>
-        dispatch(setPageSettings({ ...settings, page: pathname, searchParams: {} }))) as unknown as Action);
+        dispatch(
+          setPageSettings({
+            ...settings,
+            page: pathname,
+            searchParams: {},
+            loaded: !!settings.instantLoading,
```

Comment:
Est ce que la valeur par défaut ne devrait pas être `true` ? Puisque si on n'a rien setup, le loader ne disparaitra jamais (ce qui est le cas sur une 404 par exemple)
```suggestion
            loaded: settings.instantLoading !== false,
```

Reply:
D'ailleurs, on a besoin de mettre les 2 dispatch ?
Genre comme ça, ça marche très bien non ?
```ts
useEffect(() => {
  dispatch(
    setPageSettings({
      ...settings,
      page: pathname,
      searchParams: {},
      loaded: settings.instantLoading !== false,
    }),
  );
}, deps);
```